apps/studio/prisma/schema.prisma

```diff
@@ -26,68 +26,73 @@ model VerificationToken {
-  // @relation("PostToParent") is the name of the relation. See relation documentation below for more info.
-  parent       Post?        @relation("PostReplies", fields: [parentPostId], references: [id])
-  // The replies to this post.
-  replies      Post[]       @relation("PostReplies")
-
-  // To return `Date`s intact through the API we use transformers
-  // https://trpc.io/docs/data-transformers
-  // This is unique so it can be used for cursor-based pagination
-  createdAt DateTime  @default(now())
-  updatedAt DateTime  @default(now()) @updatedAt
-  deletedAt DateTime? @db.Timestamp
-
-  @@index([parentPostId])
-  @@index([parentPostId, deletedAt, id])
-  @@index([deletedAt])
-  @@index([createdAt])
-}
+model Resource {
+  id             String       @id @default(cuid())
+  name           String
+  siteId         String
+  parentId       String?
+  lastEditTime   DateTime?
+  lastEditUserId Int
+  blobId         String?      @unique
+  variant        ResourceType
+  userId         String
 
-model LikedPosts {
-  post      Post     @relation(fields: [postId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  postId    String
-  user      User     @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  userId    String
-  createdAt DateTime @default(now())
+  site         Site         @relation(fields: [siteId], references: [id])
+  parent       Resource?    @relation("parentRelation", fields: [parentId], references: [id])
+  child        Resource[]   @relation("parentRelation")
+  lastEditUser User?        @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction) // Don't delete the file if the user is deleted
+  blob         Blob?
+  Permission   Permission[]
 
-  @@id([postId, userId])
+  @@unique([siteId, parentId, name], name: "uniqueChildNames")
 }
 
 model User {
-  id            String       @id @default(cuid())
-  name          String?
-  username      String?      @unique
-  email         String?      @unique
-  emailVerified DateTime?
-  image         String?
-  bio           String?
-  posts         Post[]
-  likedPosts    LikedPosts[]
-  accounts      Accounts[]
+  id    String @id @default(cuid())
+  name  String
+  email String @unique
+  phone String @unique
+
+  Resource   Resource[]
+  Permission Permission[]
+}
+
+model Permission {
+  id         String   @id @default(cuid())
+  resourceId String
+  userId     String
+  role       RoleType
+
+  // Delete the row if either resource or user is deleted, default behavior cascade
+  resource Resource @relation(fields: [resourceId], references: [id])
+  user     User     @relation(fields: [userId], references: [id])
+
+  // userId and resource Id should be unique? Each user should have unambiguous access to resource? Or not if we take the strictest or laxest perm for the user
+  @@unique([userId, resourceId], name: "unambiguousPermissionsPerUserResource")
```

Comment:
I think this depends on whether the roles are strict subsets of each other, from what we discussed last week this seems to be the case, but we can drop it in future if this changes

Reply:
Removing for now

Reply:
From discussion with Chin, don't lock in to strict subset roles